### DISCUSSION TITLE
Add pull request indicator environnement variable to vercel system variables

### DISCUSSION THREAD

**Discussion by eliasm307** [vercel/vercel#6599] [Ideas]
By default vercel builds a preview for every commit on every branch. This might be useful for some but I find it wasteful so I've added a custom script in the build step to only allow a build if the commit is on the production branch and if it there are changes in a relevant package (I'm using a mono repo).

This works well however now I don't get preview links for pull requests automatically (which was useful) , because the pull request commits are counted as not being on the production branch so I need to create these manually since there is no information from the vercel environment variables to determine if a commit is part of a pull request.

I can probably setup a github action to do this but it would be good if the Vercel for Github integration just provided an environment variable saying if the preview is part of a pull request and that could be used as part of the custom build script, to keep the conditional build logic in one place.

An example implementation could be an environment variable that says "true" /"false" or an environment variable that gives the pull request name if it exits otherwise is blank if the commit is not associated with a pull request.

**Comment by leerob** [comment]
> This might be useful for some but I find it wasteful so I've added a custom script in the build step to only allow a build if the commit is on the production branch and if it there are changes in a relevant package.

Have you seen the Ignore Build Step?

https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel

**Reply by eliasm307** [reply]
Hey, yeah sorrythat's what I meant by the build script, sorry I had forgotten the name. I have created a custom one that uses the available environment variables to only build on the main branch but as mentioned there is no environment variable indicating whether a commit is part of an active pull request so I am having to manually create preview builds at the moment for pull requests.

Does this clarify what I mean?

**Reply by leerob** [reply]
> part of an active pull request

What does this mean? Are you pushing commits for closed PRs?

> manually create preview builds at the moment for pull requests.

So you're making your own custom CI/CD here using the Vercel CLI?

**Reply by eliasm307** [reply]
@leerob no im not adding commits to closed pull requests, I just meant adding commits to a branch with an active/open pull request. 

Yes im using the Vercel CLI to manually create the Vercel build previews for pull requests. 

Just to add some more information, a list of the available environment variables available to the Ignore Build script is shown [here](https://vercel-com.cdn.ampproject.org/v/s/vercel.com/docs/environment-variables.amp?usqp=mq331AQIKAGwASCAAgM%3D&amp_js_v=0.1#system-environment-variables) 

None of the currently available environment variables can be used to determine if a build is part of a pull request however the Vercel for Github integration seems to be aware of this as it comments on pull requests and provides urls to the preview build progress. 

So I was wondering if it is possible to add some pull request context to the environment variables so the ignore build scripts can account for this? 

Does this make sense?
